CMakePresets.json

```diff
@@ -0,0 +1,528 @@
+        "lhs": "${hostSystemName}",
+        "rhs": "Windows"
+      },
+      "architecture": {
+        "value": "x64",
+        "strategy": "external"
+      },
+      "cacheVariables": {
+        "CMAKE_C_COMPILER": "cl",
+        "CMAKE_CXX_COMPILER": "cl"
+      }
+    },
+    {
+      "name": "windows-msvc-debug-ootree-fast",
+      "displayName": "Windows x64: MSVC / Debug / Out of tree (fast)",
+      "inherits": "windows-config-base",
+      "binaryDir": "${sourceDir}/.build/${presetName}",
+      "installDir": "${sourceDir}/install",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Debug",
+        "RE2C_BUILD_RE2GO": false
+      }
+    },
+    {
+      "name": "windows-msvc-debug-ootree-full",
+      "displayName": "Windows x64: MSVC / Debug / Out of tree (full)",
+      "inherits": "windows-msvc-debug-ootree-fast",
+      "installDir": "${sourceDir}/install-2",
+      "cacheVariables": {
+        "RE2C_BUILD_RE2GO": true,
+        "RE2C_BUILD_LIBS": true,
+        "RE2C_REBUILD_LEXERS": true,
+        "RE2C_REBUILD_DOCS": false,
+        "RE2C_FOR_BUILD": "${sourceDir}/install/bin/re2c"
+      }
+    },
+    {
+      "name": "windows-msvc-debug-intree-fast",
+      "displayName": "Windows x64: MSVC / Debug / In tree (fast)",
+      "inherits": "windows-config-base",
+      "binaryDir": "${sourceDir}",
+      "installDir": "${sourceDir}/install",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Debug",
+        "RE2C_BUILD_RE2GO": false
+      }
+    },
+    {
+      "name": "windows-msvc-debug-intree-full",
+      "displayName": "Windows x64: MSVC / Debug / In tree (full)",
+      "inherits": "windows-msvc-debug-intree-fast",
+      "installDir": "${sourceDir}/install-2",
+      "cacheVariables": {
+        "RE2C_BUILD_RE2GO": true,
+        "RE2C_BUILD_LIBS": true,
+        "RE2C_REBUILD_LEXERS": true,
+        "RE2C_REBUILD_DOCS": false,
+        "RE2C_FOR_BUILD": "${sourceDir}/install/bin/re2c"
+      }
+    },
+    {
+      "name": "windows-msvc-release-ootree-fast",
+      "displayName": "Windows x64: MSVC / Release / Out of tree (fast)",
+      "inherits": "windows-config-base",
+      "binaryDir": "${sourceDir}/.build/${presetName}",
+      "installDir": "${sourceDir}/install",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release",
+        "RE2C_BUILD_RE2GO": false
+      }
+    },
+    {
+      "name": "windows-msvc-release-ootree-full",
+      "displayName": "Windows x64: MSVC / Release / Out of tree (full)",
+      "inherits": "windows-msvc-release-ootree-fast",
+      "installDir": "${sourceDir}/install-2",
+      "cacheVariables": {
+        "RE2C_BUILD_RE2GO": true,
+        "RE2C_BUILD_LIBS": true,
+        "RE2C_REBUILD_LEXERS": true,
+        "RE2C_REBUILD_DOCS": false,
+        "RE2C_FOR_BUILD": "${sourceDir}/install/bin/re2c"
+      }
+    }
+  ],
+  "buildPresets": [
+    {
+      "name": "linux-build-base",
+      "hidden": true,
+      "configuration": "Debug",
```

Comment:
Why do we have a default here? Isn't it better to explicitly specify in each preset, to make them symmetric? Same question for macOS and Windows, and for configurations.

Reply:
Your change is exactly what I meant --- I know it's a few more lines of code, but I'm willing to trade that off for symmetry in the presets.